pkg/config/etcd/config.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 flannel authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcd
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/flannel-io/flannel/pkg/ip"
+)
+
+type EtcdSubnetMngrConfig struct {
+	Network       ip.IP4Net
+	IPv6Network   ip.IP6Net
+	SubnetMin     ip.IP4
+	SubnetMax     ip.IP4
+	IPv6SubnetMin *ip.IP6
+	IPv6SubnetMax *ip.IP6
+	SubnetLen     uint
+	IPv6SubnetLen uint
+}
+
+func CheckConfig(esmConfig *EtcdSubnetMngrConfig, enableIPv4, enableIPv6 bool) error {
```

Comment:
I would replace this function with one called CheckConfig(....) which verifies common things for both etcd and kube subnet Manager. I think right now we can only check that there is a backend, that there is a backendType and that there is a Network.
The rest of the stuff, I would move it inside "GetNetworkConfig(ctx context.Context) (*config.Config, error)" of "type Manager interface", (probably in a different function which gets called by "GetNetworkConfig")